### ISSUE TITLE
Wrong edge length

### ISSUE THREAD

**Issue by white-gecko** [paulbrodersen/netgraph#99]
With the following code, the (2,3) 2.46 edge and the (1,2) 1.79 and (1,3) 1.55 edges should make a triangle, but the are drawn nearly parallel, this is wrong, as I see it.

```
import matplotlib.pyplot as plt

from netgraph import Graph # pip install netgraph OR conda install -c conda-forge netgraph

# right triangle
edge_length = {
    (0, 1) : 1.41,
    (0, 2) : 2.77,
    (2, 3) : 2.46,
    (0, 4) : 2.43,
    (1, 2) : 1.79,
    (1, 3) : 1.55,
    (3, 4) : 1.07,
}
edges = list(edge_length.keys())

fig, ax = plt.subplots()
Graph(edges, edge_labels=edge_length, node_layout='geometric',
      node_layout_kwargs=dict(edge_length=edge_length), ax=ax, scale=(3, 3))
ax.set_aspect('equal')
plt.show()
```

The layout should be like this:

![grafik](https://github.com/user-attachments/assets/729d23bf-2fd8-4275-bdb9-ff5e0cbde27f)

edit:

The actual output I'm referring to is:

![grafik](https://github.com/user-attachments/assets/6a7d77a0-5973-4daa-8018-0774806853ad)

after some runs, I also got, which looks similar to the original graph:

![grafik](https://github.com/user-attachments/assets/8574a031-c025-4d10-9400-265374647860)

but I also got:

![grafik](https://github.com/user-attachments/assets/06f7bc54-00c6-4d72-b645-85c1635cc581)

where the two edges (2, 3) : 2.46 and (0, 4) : 2.43 should be nearly the same length, but they are not. How can I ensure that the length is actually respected?

**Comment by paulbrodersen** [comment]
Hi, thanks for raising the issue. I understand your frustration, unfortunately there isn't much that can be done about this issue (or at least: despite my best efforts, I haven't come up with a better solution). As I note in the [documentation](https://netgraph.readthedocs.io/en/latest/node_layout.html#netgraph.get_geometric_layout): 

> Node positions are determined through non-linear optimisation: the total distance between nodes is maximised subject to the constraint imposed by the edge lengths, which are used as upper bounds. 

Optimisation procedures can and often do get stuck in local optima. The one parameter that you could tweak is the `tol` parameter. By default, its value is 0.001, but smaller values should result in additional iterations and thus hopefully improvements in the layout (at the cost of longer running times). 

```python
fig, ax = plt.subplots()
Graph(edges, edge_labels=edge_length, node_layout='geometric',
      node_layout_kwargs=dict(edge_length=edge_length, tol=0.0001), ax=ax, scale=(3, 3))
ax.set_aspect('equal')
plt.show()
```

**Comment by white-gecko** [comment]
Thank you for the explanation and the hint for the documentation. I thought the positions would be determined by sime kind of tri- or multilateration.

One idea that came to my head was to get the resulting positions of the nodes and judge on the quality of a result based on the difference between actual edge length and intended edge length to then try multiple runs until some quality is reached. (this lead me to #100 )

**Comment by white-gecko** [comment]
Your explanation resolves the issue as a requirement to your code for me.

**Comment by paulbrodersen** [comment]
> I thought the positions would be determined by sime kind of tri- or multilateration. 

When I looked into that, I got the impression that this would be computationally too demanding for medium sized graphs. 
Also, it would require the triangle inequality to be obeyed, which isn't an issue with the current implementation. 
For truly geometric graphs (i.e. street networks), this would probably be an advantage. However, there are many graphs with edges with a more general notion of "length" that would not yield to multilateration. 

> to then try multiple runs 

I thought about this approach but decided against it as this layout is already by far the slowest to compute.
